### ISSUE TITLE
[XBOX] Manual Scan detects no files in some folders

### ISSUE THREAD

**Issue by stuffu** [libretro/RetroArch#16039]
Running a manual scan (RA 1.16 Xbox Series) returns with "No valid content" in some folders, others lists the content as it should. Same settings in manual scan (just changing library and do a new scan). The permissions are identical for all folders and I haven't set a core or anything else.

Specific there are issues with two specific systems that I have found out so far. Amiga and C64 which contains mainly .lha and .d64 files. The folders that works I have not verified if they pick up every file in the folder but they didn't return an error at least.

I also tried to manually list file extensions but no difference as well as disabling/enabling scan within archive.

What could cause this? Manual scan shouldn't care about file extensions as far as I know.

**Comment by i30817** [comment]
How are you specifying the extensions in the manual scanner, `lha` or `.lha`? Only the first is supposed to work.

Are you overwriting the playlist or adding to it? Are you using the 'arcade dat filter' or 'validate existing entries' settings that prevents adding entries if they don't match either a .dat names or the database names?

Besides those traps the manual scanner has a few user annoyances, namely it's very easy to be unable to get thumbnails if you don't change the default system name to the specific console database you want (because the default, `<Content directory>` requires that the content directory be named for the system database name to connect to the thumbnails - a similar thing happens with custom).

**Comment by i30817** [comment]
I'm also not sure if RetroArch on the Xbox has a sandbox, but it's kind of easy to end up in a weird situation in android where you 'appear' not to have a path to wherever you placed your files because RetroArch, the application that is the 'user' of a sandbox doesn't have permissions to list the directory parent you're trying to use to navigate to the games.

For instance, if you try to load content in android and choose of the initial list of directories `/` the root, RetroArch will attempt to list the directories of root, and present the option "parent directory" and the error 'directory not found'.

Moreover I think you can't even choose a sdcard.

**Comment by stuffu** [comment]
@i30817 Yup, tried `lha` as well as `.lha` and also without extensions. It’s also just about creating the playlist, I can’t even generate one for some systems, like the Amiga. Thumbnails is a later issue :) 

Tried both placing the playlists on internal storage as well as external but same result.

I just can’t get it why it can’t create a basic playlist based on just the file names. I can configure cores, thumbnails etc later but need to start with a playlist.

not sure if logs will help out in this case?

**Comment by i30817** [comment]
Could be that your playlist directory is unwritable\doesnt exist.

This can happen if you reused retroarch.cfg from different installs (in android, because there are multiple versions of RetroArch android with different sandbox paths), or platforms.

Can't write the playlist if the directory to write in doesn't exist.

**Comment by stuffu** [comment]
I don’t think so, many of the playlists generates fine and they are in the same folder. Following up with a log later on…

thanks for the suggestions though!

**Comment by stuffu** [comment]
Logs don't reveal anything. Maybe manual scan results doesn't end up in the log.

**Comment by i30817** [comment]
This is going to sound a bit silly. But iirc, the Xbox might be using a very old version of RetroArch so it might be relevant.

In the path of the games that don't scan, is there a dot ('.') in either a parent directory or the name of the content scanned, that is not the extension marker? 


I don't recall how windows marked hidden folders, but for a long while RetroArch had a bug where instead of looking for a dot in the first character of a filename to look for hidden directories or files, it looked for any dot from the start of the filename (files first chopped off extensions)... or something like that.

It was 2 years ago or something, so maybe absolutely not relevant, but I have no idea how late or not the Xbox port is.
